src/util/side_metadata/mod.rs

```diff
@@ -107,6 +191,12 @@
 mod constants;
 mod global;
 mod helpers;
+// #[cfg(target_pointer_width = "32")]
+mod helpers_32;
```

Comment:
If you use `cfg!(target_pointer_width="32")`, those code will be checked at compile time in 64bits as well. But if you use `#[cfg(target_pointer_width="32")]` for blocks or functions, those code will won't be checked. Ideally 32 bits code should not be compiled for 64bits at all.